src/tools/rbd/action/Import.cc

```diff
@@ -649,7 +649,7 @@ static int do_import_header(int fd, int import_format, uint64_t &size, librbd::I
     return 0;
   }
 
-  if (fd == STDIN_FILENO || size < utils::RBD_IMAGE_BANNER_V2.size()) {
+  if (fd == STDIN_FILENO && size < utils::RBD_IMAGE_BANNER_V2.size()) {
```

Comment:
To be honest, I'm not sure why either of these checks are needed. There is no seek operations and the size isn't needed to validate the header. This check and the ```size``` parameter can probably just be dropped.